02/app.js

```diff
@@ -2,6 +2,39 @@ document.addEventListener('DOMContentLoaded', init);
 
 function init() {
     const divList = document.querySelectorAll('div');
+
+    setBorderColorAsync(divList[0], 'red', function(){})
+        .then(resp => setBorderColorAsync(divList[1], 'blue', function(){}))
+        .then(resp => setBorderColorAsync(divList[2], 'green', function(){}))
+        .then(resp => console.log('finish'))
+        .catch(err => console.error(err));
+}
+
+function setBorderColorAsync(element, color, callback) {
+       
+    if(element && element instanceof HTMLElement) {
```

Comment:
W zasadzie lepiej by było, aby ten warunek był wew. `Promise`.
Jeśli będzie nie spełniony warunek to wywołuję `reject`.
Wtedy mam "pełną" obsługę przez obietnicę :)